docs/root/configuration/http/http_filters/credential_injector_filter.rst

```diff
@@ -24,41 +20,54 @@ Configuration
 * This filter should be configured with the type URL ``type.googleapis.com/envoy.extensions.filters.http.credential_injector.v3.CredentialInjector``.
 * :ref:`v3 API reference <envoy_v3_api_msg_extensions.filters.http.credential_injector.v3.CredentialInjector>`
 
-Currently the filter supports :ref:`generic <envoy_v3_api_msg_extensions.http.injected_credentials.generic.v3.Generic>` only.
-Other credential types can be supported as extensions.
+The filter is configured with one of the following supported ``credential_injector`` extensions. Extensions are responsible for fetching the credentials
+from the source. The credentials obtained are then injected into the ``Authorization`` header of the proxied HTTP requests, utilizing either the ``Basic``
+or ``Bearer`` scheme.
+
+Generic Credential Injector
+===========================
+* This extension should be configured with the type URL ``type.googleapis.com/envoy.extensions.http.injected_credentials.generic.v3.Generic``.
+* :ref:`generic <envoy_v3_api_msg_extensions.http.injected_credentials.generic.v3.Generic>`
 
 Here is an example configuration with Generic credential, which injects an HTTP Basic Auth credential into the proxied requests.
 
-.. literalinclude:: _include/credential-injector-filter.yaml
+.. literalinclude:: _include/credential-injector-generic-filter.yaml
     :language: yaml
-    :lines: 28-41
-    :caption: :download:`credential-injector-filter.yaml <_include/credential-injector-filter.yaml>`
+    :lines: 22-35
+    :caption: :download:`credential-injector-filter.yaml <_include/credential-injector-generic-filter.yaml>`
 
 
-credential.yaml for Basic Auth:
+credential which is being used to inject a Basic Auth credential into the proxied requests:
 
-.. code-block:: yaml
-
-  resources:
-  - "@type": "type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.Secret"
-    name: credential
-    generic_secret:
-      secret:
-        inline_string: "Basic base64EncodedUsernamePassword"
+.. literalinclude:: _include/credential-injector-generic-filter.yaml
+    :language: yaml
+    :lines: 52-55
+    :caption: :download:`credential-injector-filter.yaml <_include/credential-injector-generic-filter.yaml>`
 
 It can also be configured to inject a Bearer token into the proxied requests.
 
-credential.yaml for Bearer Token:
+credential for Bearer Token:
```

Comment:
```suggestion
Credential for Bearer Token:
```

also wondering if `Bearer Token` should be literal

Reply:
updated. Thanks